clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -825,6 +825,97 @@ protected void produceToLocalKafka(
+              new PubSubTopicPartitionImpl(newSourceTopic, sourceTopicPartition.getPartitionNumber());
+          try {
+            upstreamStartOffset =
+                getTopicPartitionOffsetByKafkaURL(sourceKafkaURL, newSourceTopicPartition, rewindStartTimestamp);
+            LOGGER.info(
+                "Get upstreamStartOffset: {} for source URL: {}, topic: {}, rewind timestamp: {}",
+                upstreamStartOffset,
+                sourceKafkaURL,
+                newSourceTopicPartition,
+                rewindStartTimestamp);
+          } catch (Exception e) {
+            /**
+             * This is actually tricky. Potentially we could return a -1 value here, but this has the gotcha that if we
+             * have a non-symmetrical failure (like, lor1 can't talk to the ltx1 broker) this will result in a remote
+             * colo rewinding to a potentially non-deterministic offset when the remote becomes available again. So
+             * instead, we record the context of this call and commit it to a repair queue to rewind to a
+             * consistent place on the RT
+             *
+             * NOTE: It is possible that the outage is so long and the rewind so large that we fall off retention. If
+             * this happens we can detect and repair the inconsistency from the offline DCR validator.
+             */
+            unreachableBrokerList.add(sourceKafkaURL);
+            upstreamStartOffset = OffsetRecord.LOWEST_OFFSET;
+            LOGGER.error(
+                "Failed contacting broker {} when processing topic switch! for {}. Setting upstream start offset to {}",
+                sourceKafkaURL,
+                sourceTopicPartition,
+                upstreamStartOffset);
+            hostLevelIngestionStats.recordIngestionFailure();
+
+            // Add to repair queue
+            if (remoteIngestionRepairService != null) {
+              this.remoteIngestionRepairService.registerRepairTask(
+                  this,
+                  buildRepairTask(
+                      sourceKafkaURL.toString(),
+                      sourceTopicPartition,
+                      rewindStartTimestamp,
+                      partitionConsumptionState));
+            } else {
+              // If there isn't an available repair service, then we need to abort in order to make sure the error is
+              // propagated up
+              throw new VeniceException(
+                  String.format(
+                      "Failed contacting broker (%s) and no repair service available!  Aborting topic switch processing for %s. Setting upstream start offset to %d",
+                      sourceKafkaURL,
+                      sourceTopicPartition,
+                      upstreamStartOffset));
+            }
+          }
+        } else {
+          upstreamStartOffset = OffsetRecord.LOWEST_OFFSET;
+        }
+      }
+      upstreamOffsetsByKafkaURLs.put(sourceKafkaURL.toString(), upstreamStartOffset);
```

Comment:
For unreachable broker, it's put into the repair service already, we shouldn't populate this map with unreachable broker, right? Was this a bug?

Reply:
I think either way works. I think for now, if it is unreachable, then we will not consume from it, until it successfully subscribe to the broker, it will re-subscribe and sync the offset again. 
Personally I think it is harmless, but if you think it is good to clean up (I am not sure if there is any gotcha that I am not aware of :) ), I can also do that. What do you think?

Reply:
I just realize that we add the unreachable broker into the map, and then later remove it from the map before subscribing:
"""
// We won't attempt to resubscribe for brokers we couldn't compute an upstream offset accurately for. We'll
      // reattempt subscription later
      // Queue up repair here:
      for (CharSequence unreachableBroker: unreachableBrokerList) {
        upstreamOffsetsByKafkaURLs.remove(unreachableBroker.toString());
      }
"""
I would recommend that we try to simply the codes as much as we can; don't add unreachable brokers into the map from the beginning, and remove the above codes in both LFSIT and AASIT.

Reply:
I can do that, btw it is only AASIT.